python/lsst/obs/base/formatters/fitsExposure.py

```diff
@@ -220,12 +228,8 @@ def readFull(self, parameters=None):
         if parameters is None:
             parameters = {}
         fileDescriptor.storageClass.validateParameters(parameters)
-        try:
-            output = fileDescriptor.storageClass.pytype(fileDescriptor.location.path, **parameters)
-        except TypeError:
-            reader = ExposureFitsReader(fileDescriptor.location.path)
-            output = reader.read(**parameters)
-        return output
+        reader = self._readerClass(fileDescriptor.location.path)
```

Comment:
Is the first path never followed?

Reply:
I think it's clearer all around if we use the ExposureFitsReader etc consistently rather than using the constructor with a filename argument since then we know for a fact we are consistent with what getComponent is doing without having to guess. I don't think before the exception ever triggered and if it did then the reader.read would also fail. Didn't seem any point having both (this code path is not used for reading components which use FitsGenericFormatter).